src/routes/StackView/StackView.tsx

```diff
@@ -32,58 +53,156 @@ export const StackView = () => {
         })),
     _.isEqual
   );
-  const moderating = useAppSelector((state) => state.view.moderating);
-  const showAuthors = useAppSelector((state) => state.board.data?.showAuthors ?? true);
-  const viewer = useAppSelector((state) => state.participants!.self);
+  const column = columns.find((c) => c.id === note?.position.column);
+  const prevColumnParent = useAppSelector((state) => {
+    const prevColumns = state.columns.filter((c) => c.index < column!.index).reverse();
+    let prevStack;
+    while (prevColumns.length > 0 && !prevStack) {
+      prevStack = state.notes.filter((n) => n.position.column === prevColumns[0]?.id && n.position.stack === null).at(-1);
+      prevColumns.shift();
+    }
+    return prevStack?.id;
+  });
+  const nextColumnParent = useAppSelector((state) => {
+    const nextColumns = state.columns.slice(column!.index + 1);
+    let nextStack;
+    while (nextColumns.length > 0 && !nextStack) {
+      nextStack = state.notes.find((n) => n.position.column === nextColumns[0].id);
+      nextColumns.shift();
+    }
+    return nextStack?.id;
+  });
+  const stacksInColumn = useAppSelector((state) => state.notes.filter((n) => n.position.column === column?.id && n.position.stack === null), _.isEqual);
+  const moderating = useAppSelector((state) => state.view.moderating, _.isEqual);
+  const showAuthors = useAppSelector((state) => state.board.data?.showAuthors ?? true, _.isEqual);
+  const viewer = useAppSelector((state) => state.participants!.self, _.isEqual);
+  const presenting = moderating && (viewer.role === "MODERATOR" || viewer.role === "OWNER");
+
+  const [transitionConfig, setTransitionConfig] = useState({
+    from: {transform: "translate(0%)", position: "absolute", opacity: 0},
+    enter: {transform: "translate(0%)", position: "relative", opacity: 1},
+    leave: {
+      transform: "translate(0%)",
+      position: "absolute",
+      opacity: 0,
+    },
+    items: {
+      parent: note,
+      stack: stackedNotes,
+      avatar: author!.user.avatar,
+      authorName,
+    },
+  });
+
+  useEffect(() => {
+    if (prevNote.current && prevNote.current?.id !== note?.id) {
+      let direction: "left" | "right" | undefined;
+      if (prevNote && prevNote.current?.position?.column === note?.position?.column) {
+        direction = prevNote.current?.position?.rank > note!.position.rank ? "right" : "left";
```

Comment:
You could move this into the else if below so

```js
if (prevnote) {
  if (prevnote.current... === note?...) {
    direction = ...
  } else {
    const old...
    ...
  }
```

Reply:
^ for me this improves readability, if you disagree just ignore this. 😄

Reply:
I just removed the check for prevnote because it is obsolete. prevnote used to be a state, but I changed it to a ref so now it is always truthy